tools/build_defs/repo/git_worker.bzl

```diff
@@ -126,7 +126,7 @@ def _update(ctx, git_repo):
 
 def init(ctx, git_repo):
     cl = ["git", "init", str(git_repo.directory)]
-    st = ctx.execute(cl, environment = ctx.os.environ)
+    st = ctx.execute(_strip_git_vars(cl), environment = ctx.os.environ)
```

Comment:
Can you remove those env vars from ctx.os.environ instead?

Reply:
I think that you can only force variables to new values (e.g. the empty value), but can't remove them. Or maybe that works if they are set to `None`? If not, that would probably be a reasonable feature request :⁠-⁠)

Reply:
@fmeum is correct. This was my first thought, but setting to the empty string upsets Git, and setting to None upsets Bazel. And yes, being able to unset a variable with None would be a great FR!

Also, I would have liked to have been able to unset a variable with `--repo_env`.

Reply:
the problem (as indicated by the test results) is that this doesn't work on Windows... We might need to implement that FR first :/

Reply:
Yeah, I saw the Windows failures. I'll dig into that and see if this can reasonably be solved within this repo rule.

Reply:
FYI, I sent https://github.com/bazelbuild/bazel/pull/24221, so it may be more efficient to wait for that.

Reply:
Thanks for that! This is definitely much cleaner. However, I still need a solution for Bazel 7. So I will continue to get this working in Windows.

Perhaps this PR (once I fix Windows) could go in and be cherry-picked into the 7 branch, then a follow-up could replace the solution based on your PR? I'm not sure what the process is for point releases.

Reply:
Yes, if there is ever a 7.5 release, we could just cherry-pick both commits.

Reply:
I have rebased and changed my PR to use your new feature. I haven't been able to get my Windows build of Bazel to pass that failing test even at HEAD, so I'm going to lean on the merge checks here. 🤞